seatunnel-connectors-v2/connector-maxcompute/src/main/java/org/apache/seatunnel/connectors/seatunnel/maxcompute/util/FormatterContext.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.maxcompute.util;
+
+import org.apache.seatunnel.common.exception.CommonError;
+import org.apache.seatunnel.common.utils.DateTimeUtils;
+
+import java.time.LocalDateTime;
+
+public class FormatterContext {
+    private final DateTimeUtils.Formatter localDateTimeFormat;
+
+    public FormatterContext(DateTimeUtils.Formatter localDateTimeFormat) {
+        String format = localDateTimeFormat.getValue();
+        validateFormat(format);
+        this.localDateTimeFormat = localDateTimeFormat;
+    }
+
+    public FormatterContext(String localDateTimeFormat) {
+        validateFormat(localDateTimeFormat);
+        this.localDateTimeFormat = DateTimeUtils.Formatter.parse(localDateTimeFormat);
+    }
+
+    public boolean isDateTimeType(Object field) {
+        return field instanceof LocalDateTime;
+    }
+
+    public String formatDateTime(Object field) {
+        if (field instanceof LocalDateTime) {
+            return this.format(((LocalDateTime) field));
+        }
+        throw CommonError.illegalArgument(
+                field.getClass().getName(),
+                "Cannot format the given value: not a LocalDateTime instance.");
+    }
+
+    private void validateFormat(String localDateTimeFormat) {
+        if (localDateTimeFormat.contains("T")) {
+            throw CommonError.illegalArgument(
+                    localDateTimeFormat,
+                    "MaxCompute does not support datetime format containing 'T'.");
+        }
+    }
```

Comment:
we should not validate the format. If user want the value contains `T`, keep it.